vllm/model_executor/guided_decoding/xgrammar_decoding.py

```diff
@@ -246,14 +218,23 @@ def from_guided_params(cls,
                 raise ValueError(str(err)) from err
 
             return cls(grammar_str=grammar_str,
-                       vocab_size=model_config.hf_text_config.vocab_size,
                        tokenizer_hash=tokenizer_hash,
                        max_threads=max_threads,
                        tokenizer_data=tokenizer_data)
+        elif guided_params.regex:
+            try:
+                xgr.Grammar.from_regex(guided_params.regex)
```

Comment:
I'm fine with leaving this in since I've wanted this in for a while and would like it over with (had a PR before you included it in this one as you know)

but ideally it's best to keep PRs focused on one thing. This mixes a feature with a bug fix